### PR TITLE
[LiveLogger] Add code to high priority messages

### LINKED ISSUE #8393: Live logger trims the message ID for SDK messages
The console logger includes NETSDK1057 but the live logger doesn't. Not sure if it's all messages or just this one. Including the message ID is useful when searching for messages from customer reports. 

Warnings work correctly. 
Haven't tried errors.

image

image

### PR DESCRIPTION
**Pull request by edvilme**
Fixes #8393 

### Context
Currently the live logger does not log the message id's. Including the message ID is useful when searching for messages from customer reports.

### Changes Made
When receiving `BuildMessageEventArgs`, the `Code` is now also passed to `MessageNode` and printed on the display.

### CODE REVIEW

File: src/MSBuild/LiveLogger/MessageNode.cs [hunk 1]
```diff
@@ -68,7 +69,7 @@ public string ToANSIString()
                         ANSIBuilder.Formatting.ForegroundColor.Red)}";
                 case MessageType.HighPriorityMessage:
                 default:
-                    return $"ℹ️ {ANSIBuilder.Formatting.Italic(Message)}";
+                    return $"ℹ️ Message {Code}: {ANSIBuilder.Formatting.Italic(Message)}";
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/MessageNode.cs:72]: Changing this unconditionally means every message now has a colon in it:

image

I think it'd be better to add the code+colon only if a code exists.

**Comment by edvilme** [comment]
Oh I agree. Should it still say message and then a colon after that for those that don't have codes?

File: src/MSBuild/LiveLogger/MessageNode.cs (hunk 1 above)
Comment (edvilme) [reply src/MSBuild/LiveLogger/MessageNode.cs:72]: Oh I agree. Should it still say message and then a colon after that for those that don't have codes?

Comment (rainersigwald) [reply src/MSBuild/LiveLogger/MessageNode.cs:72]: IMO saying "message" is unnecessary; we're indicating that through formatting and the ℹ️

File: src/MSBuild/LiveLogger/MessageNode.cs [hunk 2]
```diff
@@ -69,7 +69,7 @@ public string ToANSIString()
                         ANSIBuilder.Formatting.ForegroundColor.Red)}";
                 case MessageType.HighPriorityMessage:
                 default:
-                    return $"ℹ️ Message {Code}: {ANSIBuilder.Formatting.Italic(Message)}";
+                    return $"ℹ️ {(Code != null ? Code + ": " : string.Empty)} {ANSIBuilder.Formatting.Italic(Message)}";
```
Comment (rainersigwald) [inline src/MSBuild/LiveLogger/MessageNode.cs:72]: This requires concatenating a string and breaks out of some optimizations (on .NET Core). If you write it a bit oddly you can get them back though:

```suggestion
                    return $"ℹ️ {Code}{(Code is not null ? ": " : string.Empty)} {ANSIBuilder.Formatting.Italic(Message)}";
```

https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ+TgAxUQAmMA3PYxi6QCyMAM4iAhgHNuvBiSatKEAHZhxAFxEAKAJS1ZjOQHZSAEgBEgTkJA8H+0tSzqQCEAXlLKArgBsvpAPxsjgDUpOYgoaTh/MwAdACiALYADuoAnjoAvrTCYlIwGeY8BIYZBgby0aQA6jAAllCcUABCHuoAMqIiAII+EGC6+sWGfCYWNrQO+TT2HDCktSLuEOru3r4BYRFRLPHJaZnZnXkFRSWEGUA== shows the compiled differences.

Comment (edvilme) [reply src/MSBuild/LiveLogger/MessageNode.cs]: That looks smart (and also cleaner), thx!

**Review by JanKrivanek — APPROVED** [review]

**Review by rainersigwald — APPROVED** [review]

**Pull request merged**